python/pyproject.toml

```diff
@@ -20,16 +20,24 @@ dependencies = [
 ]
 
 [project.optional-dependencies]
+torch = ["torch"]
 srt = ["aiohttp", "decord", "fastapi", "hf_transfer", "huggingface_hub", "interegular",
        "packaging", "pillow", "psutil", "pydantic", "python-multipart",
-       "torch", "torchao", "uvicorn", "uvloop", "zmq",
-       "vllm==0.5.5", "outlines>=0.0.44", "modelscope"]
+       "torchao", "uvicorn", "uvloop", "zmq",
+       "outlines>=0.0.44", "modelscope"]
+
+#xpu is not enabled in public vllm and torch whl,
+#need to follow https://docs.vllm.ai/en/latest/getting_started/xpu-installation.htmlinstall vllm
+vllm = ["vllm==0.5.5"]
```

Comment:
vllm and torch should be included in srt. A better approach is
runtime_common = ["aiohttp", ... "modelscope"]
srt = ["sglang[runtime_common]", "torch", "vllm"]
srt_xpu = ["sglang[runtime_common]"]

Reply:
Resolved